### DISCUSSION TITLE
How to save various old versions of the lol clients with minimal disk occupation to replay different versions of playback files?

### DISCUSSION THREAD

**Discussion by JerryLiew** [fraxiinus/ReplayBook#119] [Feature Request]
First of all, the question I want to ask is, what folders/files  are required in the lol client if I want to replay the rofl? and then lets discuss how to minimize the Required files. then  we can store all of the old version of lol client and rofl files  with minimal disk occupation , and we can reply it everytime  even after a  long time.  thanks a lot. 

( by the way, I'm very happy to find this tool. I like to save all my game records for future viewing, but I used to do it by recording the screen. It's really cumbersome  and takes up a lot of hard disk. LOL ,i will share this toold with my friends, and i am also a programer, hope i can make my contribution for this tool in the future.  appreciate)

**Comment by OrionCodeDev** [comment]
Welcome to Riot Games. It's not a Valve with CS:GO where you can replay any recorded file.

**Comment by fraxiinus** [comment]
I haven't done sufficient testing to know what exact files are required. I do know that you only need the "GAME" folder, you don't need the client.

I haven't been able to try it yet, but this project: [preyneyv/league-vcs](https://github.com/preyneyv/league-vcs) aims to reduce file sizes of backups by only saving the difference between patches. Maybe give it a shot?

**Reply by JerryLiew** [reply]
thanks. It looks like a good stuff,and  i am using that now,  but I won't know how it performs before lol is updated. if it performs well.i wll consider migrating it into replaybook as an optional option.

**Reply by fraxiinus** [reply]
Sounds great! The author for league-vcs asked if I wanted to add integration to ReplayBook, and I am 100% down. I'm still resting after releasing 1.4.0, so I haven't started work yet, but league-vcs integration will be the headline feature for version 1.5.0

**Reply by JerryLiew** [reply]
I have a new question: should runes and icons be backed up with the client together to avoid loss them after update? I noticed that what you are doing now is download them from a website. Is the website reliable? In my opinion, only when you are in full control can you be safe, much less  these things take up very little storage space.

**Reply by fraxiinus** [reply]
I get the rune images directly from Riot's data dragon API. That is probably the most reliable source at the moment. It allows getting images from specific versions as well, but that feature is not being used in ReplayBook.

Runes are cached locally once they are downloaded. So if you looked at an old replay with kleptomancy, that image stays in cache. So even if you patch league of legends, that kleptomancy image will remain the ReplayBook's image cache.

**Reply by JerryLiew** [reply]
Good, i'm now preparing for my postgraduate examination in a month. After that, i'll try to make my contributions for this.

**Reply by fraxiinus** [reply]
Good luck on your post graduate exam!

Update: seems I'm wrong on the runes thing. With the latest patch some runes will appear as "N/A" if they are no longer in the current patch. This is a problem stemming from the "generate_runes.py" script which downloads the rune indexes that ReplayBook uses. To fix this the script would somehow need to be able to include all runes that have ever existed in the index. Not sure how to accomplish this at the moment. Perhaps a list of patches when runes were changed, and load those? Would require more research.

**Reply by JerryLiew** [reply]
i have wrote a tools  like 
[League-vcs](https://github.com/preyneyv/league-vcs) in C#, but i am not sure how to make a intergration with replaybook.

**Reply by JerryLiew** [reply]
Maybe some useful information here https://github.com/preyneyv/league-vcs/issues/1

**Reply by fraxiinus** [reply]
it would require a lot of changes to replaybook, if you could link your C# implementation of league-vcs I can take a look. Or maybe add it as a project to ReplayBook.sln and create a PR

**Reply by JerryLiew** [reply]
I have found some rules to reduce the patch size between two different version. but  I only tested between 12.23 and 12.24. More test in different versions are required for making sure the rules correct. so i have to wait for the game updates, and test again.

**Reply by JerryLiew** [reply]
hi, lol client has been updated today.so i tested my tool again, it worked well. In my tests which based on three consecutive chinese special edition patches, it can reduce about 37% size of a whole client.but i dont know how it performes in other area's version. by the way, the whole  VCS tool was created with about only 100 lines of code. i wrote a winform program to do that. if it possible, maybe you can give it a shot to test the english version, i will upload the souce code to github  tomorrow.

**Reply by JerryLiew** [reply]
https://github.com/JerryLiew/LOL-VCS/tree/master

**Reply by fraxiinus** [reply]
Thanks for sharing, I'll give it a shot and then try integration

**Comment by fraxiinus** [comment]
You can track the progress of League-VCS support at this issue: #160 

If you have suggestions on how it should work, or features you would like, please post in the issue.

**Reply by JerryLiew** [reply]
i will, thanks for your working.

**Comment by JerryLiew** [comment]
Hi, finally I implemented a new tool to do this after a long time,Now it's possible to save client files in a smaller size
[LeagueBackuppper]https://github.com/JerryLiew/LeagueBackupper

**Reply by fraxiinus** [reply]
Amazing! The file size reduction is really impressive!
